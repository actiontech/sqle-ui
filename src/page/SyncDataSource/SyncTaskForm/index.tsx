import { FormInstance } from 'antd';
import { IInstanceTaskDetailResV1 } from '../../../api/common';
import SyncTaskForm from './SyncTaskForm';

export type SyncTaskFormFields = {
  source: string;
  version: string;
  url: string;
  instanceType: string;
  ruleTemplateName: string;
  syncInterval: string;
};

export type SyncTaskFormProps = {
  submit: (values: SyncTaskFormFields) => Promise<void>;
  form: FormInstance<SyncTaskFormFields>;
  defaultValue?: IInstanceTaskDetailResV1;
};

export default SyncTaskForm;
