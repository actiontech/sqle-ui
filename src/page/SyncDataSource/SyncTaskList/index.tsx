import { SyncOutlined } from '@ant-design/icons';
import { useRequest } from 'ahooks';
import { Button, Card, message, Space, Table } from 'antd';
import { useTranslation } from 'react-i18next';
import sync_instance from '../../../api/sync_instance';
import { ResponseCode } from '../../../data/common';
import { SyncTaskListTableColumnFactory } from './column';
import { Link } from '../../../components/Link';

const SyncTaskList: React.FC = () => {
  const { t } = useTranslation();

  const syncAction = (taskId: string) => {
    const hideLoading = message.loading(
      t('syncDataSource.syncTaskList.syncTaskLoading')
    );
    sync_instance
      .triggerSyncInstanceV1({
        task_id: taskId,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(t('syncDataSource.syncTaskList.syncTaskSuccessTips'));
          refresh();
        }
      })
      .finally(() => {
        hideLoading();
      });
  };
  const deleteAction = (taskId: string) => {
    const hideLoading = message.loading(
      t('syncDataSource.syncTaskList.deleteTaskLoading')
    );
    sync_instance
      .deleteSyncInstanceTaskV1({
        task_id: taskId,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(
            t('syncDataSource.syncTaskList.deleteTaskSuccessTips')
          );
          refresh();
        }
      })
      .finally(() => {
        hideLoading();
      });
  };

  const { loading, data, refresh } = useRequest(() =>
    sync_instance
      .GetSyncInstanceTaskList()
      .then((res) => ({ list: res.data?.data ?? [] }))
  );

  return (
    <Card
      title={
        <Space>
          {t('syncDataSource.syncTaskList.title')}
          <Button onClick={refresh} data-testid="refreshTable">
            <SyncOutlined spin={loading} />
          </Button>
        </Space>
      }
      extra={[
        <Button type="primary" key="addSyncTask">
          <Link to="syncDataSource/create">
            {t('syncDataSource.syncTaskList.addSyncTask')}
          </Link>
        </Button>,
      ]}
    >
      <Table
        rowKey="id"
        dataSource={data?.list ?? []}
        loading={loading}
        pagination={false}
        columns={SyncTaskListTableColumnFactory(syncAction, deleteAction)}
      />
    </Card>
  );
};

export default SyncTaskList;
