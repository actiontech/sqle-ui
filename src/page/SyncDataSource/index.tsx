import { Card, PageHeader, Typography } from 'antd';
import { useTranslation } from 'react-i18next';
import { Outlet } from 'react-router-dom';

const SyncDataSource: React.FC = () => {
  const { t } = useTranslation();

  return (
    <>
      {/* IFTRUE_isCE */}
      <section className="padding-content">
        <Card>
          {t('syncDataSource.ceTips')}
          <Typography.Paragraph>
            <ul>
              <li>
                <a href="https://actiontech.github.io/sqle-docs-cn/">
                  https://actiontech.github.io/sqle-docs-cn/
                </a>
              </li>
              <li>
                <a href="https://www.actionsky.com/">
                  https://www.actionsky.com/
                </a>
              </li>
            </ul>
          </Typography.Paragraph>
        </Card>
      </section>
      {/* FITRUE_isCE */}

      {/* IFTRUE_isEE */}
      <article>
        <PageHeader title={t('syncDataSource.pageTitle')} ghost={false}>
          {t('syncDataSource.pageDesc')}
        </PageHeader>
        <section className="padding-content">
          <Outlet />
        </section>
      </article>
      {/* FITRUE_isEE */}
    </>
  );
};

export default SyncDataSource;
