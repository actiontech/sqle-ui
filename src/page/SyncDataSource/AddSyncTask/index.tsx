import { useBoolean } from 'ahooks';
import { Button, Card, Modal, Result } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import sync_instance from '../../../api/sync_instance';
import { ICreateSyncInstanceTaskV1Params } from '../../../api/sync_instance/index.d';
import BackButton from '../../../components/BackButton';
import { ResponseCode } from '../../../data/common';
import EmitterKey from '../../../data/EmitterKey';
import EventEmitter from '../../../utils/EventEmitter';
import SyncTaskForm, { SyncTaskFormFields } from '../SyncTaskForm';
import { Link } from '../../../components/Link';

const AddSyncTask: React.FC = () => {
  const { t } = useTranslation();
  const [form] = useForm<SyncTaskFormFields>();
  const [
    resultModalVisibility,
    { setTrue: openResultModal, setFalse: closeResultModal },
  ] = useBoolean();

  const submit = (values: SyncTaskFormFields) => {
    const params: ICreateSyncInstanceTaskV1Params = {
      db_type: values.instanceType,
      global_rule_template: values.ruleTemplateName,
      source: values.source,
      sync_instance_interval: values.syncInterval,
      url: values.url,
      version: values.version,
    };
    return sync_instance.createSyncInstanceTaskV1(params).then((res) => {
      if (res.data.code === ResponseCode.SUCCESS) {
        openResultModal();
      }
    });
  };

  const resetAndCloseResultModal = useCallback(() => {
    closeResultModal();
    form.resetFields();
    EventEmitter.emit(EmitterKey.Refresh_Sync_Task_Rule_Template_Tips);
  }, [closeResultModal, form]);

  return (
    <Card
      title={t('syncDataSource.addSyncTask.title')}
      extra={[<BackButton key="goBack" />]}
    >
      <SyncTaskForm submit={submit} form={form} />

      <Modal
        title={t('common.operateSuccess')}
        footer={null}
        closable={false}
        open={resultModalVisibility}
      >
        <Result
          status="success"
          title={t('syncDataSource.addSyncTask.successTips')}
          subTitle={
            <Link to={`syncDataSource`}>
              {t('syncDataSource.addSyncTask.successGuide')} {'>'}
            </Link>
          }
          extra={[
            <Button key="close" onClick={closeResultModal}>
              {t('common.close')}
            </Button>,
            <Button
              type="primary"
              key="resetAndClose"
              onClick={resetAndCloseResultModal}
            >
              {t('common.resetAndClose')}
            </Button>,
          ]}
        />
      </Modal>
    </Card>
  );
};

export default AddSyncTask;
