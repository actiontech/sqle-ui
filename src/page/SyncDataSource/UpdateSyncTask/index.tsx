import { useBoolean } from 'ahooks';
import { Button, Card, Empty, message, Spin, Typography } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import { useCallback, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useParams } from 'react-router-dom';
import { IInstanceTaskDetailResV1 } from '../../../api/common';
import sync_instance from '../../../api/sync_instance';
import { IUpdateSyncInstanceTaskV1Params } from '../../../api/sync_instance/index.d';
import BackButton from '../../../components/BackButton';
import EmptyBox from '../../../components/EmptyBox';
import { ResponseCode } from '../../../data/common';
import SyncTaskForm, { SyncTaskFormFields } from '../SyncTaskForm';

const UpdateSyncTask: React.FC = () => {
  const { t } = useTranslation();
  const [form] = useForm<SyncTaskFormFields>();
  const { taskId } = useParams<{ taskId: string }>();
  const [initError, setInitError] = useState('');
  const history = useHistory();
  const [retryLoading, { toggle: setRetryLoading }] = useBoolean(false);
  const [finishGetSyncInstanceTask, { toggle: setFinishGetSyncInstanceTask }] =
    useBoolean(false);
  const [syncInstanceTask, setSyncInstanceTask] = useState<
    IInstanceTaskDetailResV1 | undefined
  >();

  const submit = (values: SyncTaskFormFields) => {
    const params: IUpdateSyncInstanceTaskV1Params = {
      task_id: taskId,
      global_rule_template: values.ruleTemplateName,
      sync_instance_interval: values.syncInterval,
      url: values.url,
      version: values.version,
    };
    return sync_instance.updateSyncInstanceTaskV1(params).then((res) => {
      if (res.data.code === ResponseCode.SUCCESS) {
        message.success(t('syncDataSource.updateSyncTask.successTips'));
        history.replace(`/syncDataSource`);
      }
    });
  };
  const getSyncInstanceTask = useCallback(() => {
    setRetryLoading(true);
    sync_instance
      .GetSyncInstanceTask({ task_id: taskId })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          setSyncInstanceTask(res.data.data);
          setInitError('');
        } else {
          setInitError(res.data.message ?? t('common.unknownError'));
        }
      })
      .finally(() => {
        setRetryLoading(false);
        setFinishGetSyncInstanceTask(true);
      });
  }, [setFinishGetSyncInstanceTask, setRetryLoading, t, taskId]);

  useEffect(() => {
    getSyncInstanceTask();
  }, [getSyncInstanceTask]);

  return (
    <Card
      title={t('syncDataSource.updateSyncTask.title')}
      extra={[<BackButton key="goBack" />]}
    >
      <EmptyBox
        if={!initError}
        defaultNode={
          <Empty
            image={Empty.PRESENTED_IMAGE_DEFAULT}
            description={
              <Typography.Text type="danger">
                {t('syncDataSource.updateSyncTask.getSyncInstanceTaskError')}:{' '}
                {initError}
              </Typography.Text>
            }
          >
            <Button
              type="primary"
              onClick={getSyncInstanceTask}
              loading={retryLoading}
            >
              {t('common.retry')}
            </Button>
          </Empty>
        }
      >
        <EmptyBox
          if={finishGetSyncInstanceTask}
          defaultNode={<Spin delay={400} />}
        >
          <SyncTaskForm
            defaultValue={syncInstanceTask}
            submit={submit}
            form={form}
          />
        </EmptyBox>
      </EmptyBox>
    </Card>
  );
};

export default UpdateSyncTask;
