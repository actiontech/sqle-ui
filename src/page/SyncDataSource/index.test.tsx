import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SyncDataSource from '.';

describe('test SyncDataSource', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<SyncDataSource />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
