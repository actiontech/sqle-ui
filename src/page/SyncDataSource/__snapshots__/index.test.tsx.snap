// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SyncDataSource should match snapshot 1`] = `
<Fragment>
  <section
    className="padding-content"
  >
    <ForwardRef>
      syncDataSource.ceTips
      <ForwardRef>
        <ul>
          <li>
            <a
              href="https://actiontech.github.io/sqle-docs-cn/"
            >
              https://actiontech.github.io/sqle-docs-cn/
            </a>
          </li>
          <li>
            <a
              href="https://www.actionsky.com/"
            >
              https://www.actionsky.com/
            </a>
          </li>
        </ul>
      </ForwardRef>
    </ForwardRef>
  </section>
  <article>
    <PageHeader
      ghost={false}
      title="syncDataSource.pageTitle"
    >
      syncDataSource.pageDesc
    </PageHeader>
    <section
      className="padding-content"
    >
      <Outlet />
    </section>
  </article>
</Fragment>
`;
